### DISCUSSION TITLE
Release scheme

### DISCUSSION THREAD

**Discussion by Blebowski** [The-OpenROAD-Project/OpenROAD#3856] [General]
Hello,

I see there is a lot of development going on. I am wondering whether there are any opinions/plans about introducing a release scheme (e.g. once a month or quarterly, or after major feature changes / enhancements) ? Maintaining something like `CHANGES.md` or `NEWS.md` would be also very handy. Once user does not follow the development almost day-to-day, it is easy to get lost.

**Comment by QuantamHD** [comment]
I would be in favor of a news/changes file, but I'm a little bit more hesitant on slow releases. 

Precision Innovations provides a daily release https://github.com/Precision-Innovations/OpenROAD/releases which I feel like is a good cadence given the fast pace of developement.

I think slower stable releases make sense when you have an entity that's willing to cherry pick bug fixes into older releases, which we don't have yet. I feel like if we embraced a slower release process at this point in time the net result would be people stuck with bugs for longer.

**Comment by mithro** [comment]
A monthly newsletter might be a good way to let people see the improvements.

**Reply by maliberty** [reply]
@tspyrou FYI

**Reply by QuantamHD** [reply]
I think two good examples of this are

The Dolphin Blog -- https://dolphin-emu.org/blog/tags/progress%20report/
SerenityOS Monthly Videos -- https://www.youtube.com/watch?v=xutO05N3_BM

**Reply by Blebowski** [reply]
Thanks @QuantamHD , I had no idea there are daily releases. I don't think Newsletter is necessary, simple "Release Notes" or "NEWS.md" would be sufficient.

**Comment by dralabeing** [comment]
Daily releases are good to get the best , updated and stable versions especially with prebuilt binaries from the Precision releases.  I do think it would be good to add some sort of updates that are not yet in the docs or clear in the commit histories. For e.g mpl2 works well but this is not yet reflected in the default flow or docs but can be flagged in an Updates.md for users to try out --somewhat like a beta version. Similar any new features that are ready for use (partitioning, pin swapping,  gate cloning etc.) can be mentioned along with a special release tag .
Monthly update frequency should be good as in special cases when an update is done sooner on a major feature.

**Comment by rovinski** [comment]
I really like the [KLayout release notes](https://www.klayout.de/development.html) which have a list of PRs in that release (with hyperlinks) and a one-line description, as well as classifying as a bugfix or enhancement. It doesn't sound like we're moving from daily releases, but it could form a semi-automatically generated monthly newsletter or similar.

**Reply by dralabeing** [reply]
This format looks really good. We discussed creating this automatically and should have this implemented shortly. @vvbandeira  @luarss
